### ISSUE TITLE
 hGetContents: illegal operation

### ISSUE THREAD

**Issue by SuperSandro2000** [maralorn/nix-output-monitor#23]
```
IdnaTest.txt line 2251: 'B; xn--1shy52abz3f.xn--7ib268q; [B1 B2 B3 V6]; [B1 B2 B3 V6]' ... ok
runTest (tests.test_idna_uts46.TestIdnaTest)
IdnaTest.txt line 2252: 'B; xn--0ugx0px1izu2h.xn--7ib268q; [B1 B2 B3 C1 V6]; [B1 B2 B3 C1 V6]' ... ok
runTest (tests.test_idna_uts46.TestIdnaTest)
IdnaTest.txt line 2253: 'B; xn--1nd362hy16e.xn--7ib433c; [B1 B2 B3 V6]; [B1 B2 B3 V6]' ... ok
runTest (tests.test_idna_uts46.TestIdnaTest)
IdnaTest.txt line 2254: 'B; xn--1nd159ecmd785k.xn--7ib433c; [B1 B2 B3 C1 V6]; [B1 B2 B3 C1 V6]' ... ok
runTest (tests.test_idna_uts46.TestIdnaTest)
IdnaTest.txt line 2255: 'T; \\u07E1\\u200C。--⸬; [B1 B3 C1 V3]; [B1 V3]' ... skipped 'Test requires support for a newer version of Unicode than this Python supports'
runTest (tests.test_idna_uts46.TestIdnaTest)
┏━ Currently building:
┃ ▶                       libxslt-1.1.34                     on  ssh://clara.r ⏲ 00:00:00 (∅00:00:36)
┃ ▶                        itstool-2.0.6                     on   ssh://rose.r ⏲ 00:00:00 (∅00:00:02)
┃ ▶             python3.8-pycparser-2.20 ⏲ 00:00:03 (∅00:00:04)
┃ ▶          python3.8-pycodestyle-2.6.0                     on  ssh://server7 ⏲ 00:00:03 (∅00:01:59)
┃ ▶               python3.8-pyasn1-0.4.8                     on  ssh://server7 ⏲ 00:00:04
┃ ▶           python3.8-ptyprocess-0.6.0                     on   ssh://rose.r ⏲ 00:00:05 (∅00:00:08)
┃ ▶              python3.8-pretend-1.0.9                     on  ssh://server7 ⏲ 00:00:06
┃ ▶                  python3.8-pbr-5.5.1                     on  ssh://server7 ⏲ 00:00:07 (∅00:01:05)
┃ ▶                  python3.8-idna-2.10                     on ssh://martha.r ⏲ 00:00:10 (∅00:00:18)
┃ ▶              python3.8-docutils-0.16                     on  ssh://server7 ⏲ 00:00:11 (∅00:00:58)
┃ ▶ python3.8-cryptography_vectors-3.4.6                     on  ssh://clara.r ⏲ 00:00:13 (∅00:00:20)
┃ ▶             python3.8-coverage-5.3.1                     on  ssh://server7 ⏲ 00:00:13
┃ ▶                glibc-locales-2.32-37                     on  ssh://clara.r ⏲ 00:08:13
┣━ Failed builds:
┃ ⚠ 1848.patch   on ssh://rose.r failed with exit code 1 after ⏲ 00:00:03
┣━━━ Builds               │ Downloads    │ Uploads │ Host
┃    ▶  1 │ ✔  68 │       │       │      │         │ localhost
┃         │       │       │ ⬇   6 │      │         │ http://amy:5000
┃         │       │       │ ⬇ 107 │      │         │ http://clara:5000
┃         │       │       │ ⬇  14 │      │         │ http://donna:5000
┃         │       │       │ ⬇  82 │      │         │ http://martha:5000
┃         │       │       │ ⬇  80 │      │         │ http://rose:5000
┃         │       │       │ ⬇ 390 │      │         │ https://cache.nixos.org
┃    ▶  3 │ ✔  69 │       │ ⬇ 136 │      │ ⬆    21 │ ssh://clara.r
┃    ▶  1 │ ✔  52 │       │ ⬇ 122 │      │ ⬆    22 │ ssh://martha.r
┃    ▶  2 │ ✔  81 │       │ ⬇ 145 │      │ ⬆    26 │ ssh://rose.r
┃    ▶  6 │ ✔  63 │       │ ⬇ 119 │      │ ⬆   123 │ ssh://server7
┗━ ∑ ▶ 13 │ ✔ 333 │ ⏳ 34 │ ⬇ 937 │ ⏳ 1 │         │ 🏁 man-db-2.9.4 ⏲ 01:48:48
nom: <stdin>: hGetContents: illegal operation (handle is semi-closed)
```

**Comment by maralorn** [comment]
Wow, you are a power user.^^

Is this with the newest version of nom?

Because I have written error handling code of which I thought it would deal with situations like this.

**Comment by SuperSandro2000** [comment]
> Wow, you are a power user.^^

I am just building the stdenv with uutils-coreutils which is not perfect yet.


I did a home-manager switch since this log but the nom version shouldn't have changed since then.

```
nom --version
nix-output-monitor 1.0.3.0
```

1.0.3.0 should be the latest version.

**Comment by maralorn** [comment]
Okay. So, the error happened only when the nix process exited anyways?

How often has this happened to you?

You write "after long running command exited". Can you pinpoint that a little more? How long is "long"? I wonder how this would have any effect …

**Comment by s1341** [comment]
I'm getting this issue immediately with a multi-user install of nix on top of ubuntu. nom is not functional at all.

**Comment by maralorn** [comment]
@s1341 You mean, it crashes immediately in the moment it get‘s the first input?

**Comment by s1341** [comment]
Yup.

**Comment by maralorn** [comment]
This issue should hopefully be fixed in 1.1.0.0. Please reopen if it isn’t.
